delfin/drivers/fujitsu/eternus/eternus_stor.py

```diff
@@ -352,3 +391,527 @@ def parse_alert(context, alert):
+                             consts.HOST_FC_ENCAPSULATE_DATA_TOTAL,
+                             consts.HOST_NAME_COUNT, consts.HOST_TYPE_COUNT)
+        self.get_iscsi_host(host_list, host_status)
+        self.get_fc_sas_host(host_list, host_status,
+                             consts.GET_HOST_SAS_ADDRESSES,
+                             consts.HOST_SAS_ENCAPSULATE_DATA_TOTAL,
+                             consts.HOST_SAS_NAME, consts.HOST_SAS_OS)
+        return host_list
+
+    def get_fc_sas_host(self, host_list, host_status, command,
+                        encapsulate_data_total, name_count, type_count):
+        host_fc_list = self.get_data(command)
+        for host_fc in host_fc_list:
+            if len(host_fc) < encapsulate_data_total:
+                continue
+            fc_name = host_fc[name_count]
+            os = host_fc[type_count].lower()
+            state = host_status.get(fc_name)
+            host_d = {
+                "name": fc_name,
+                "storage_id": self.storage_id,
+                "native_storage_host_id": fc_name,
+                "os_type": consts.HOST_OS_TYPES_MAP.get(
+                    os, constants.HostOSTypes.UNKNOWN),
+                "status": constants.HostStatus.NORMAL if
+                state is not None and state ==
+                consts.HOST_PATH_STATUS_SPECIFIC_TWO else
+                constants.HostStatus.OFFLINE
+            }
+            host_list.append(host_d)
+
+    def get_iscsi_host(self, host_list, host_status):
+        host_iscsi_list = self.get_iscsi_host_data()
+        for host_iscsi in host_iscsi_list:
+            iscsi_name = host_iscsi.get('name')
+            state = host_status.get(iscsi_name)
+            os = host_iscsi.get('os')
+            os = os.lower() if os else None
+            host_d = {
+                "name": iscsi_name,
+                "storage_id": self.storage_id,
+                "native_storage_host_id": iscsi_name,
+                "os_type": consts.HOST_OS_TYPES_MAP.get(
+                    os, constants.HostOSTypes.UNKNOWN),
+                "status": constants.HostStatus.NORMAL if
+                state is not None and state ==
+                consts.HOST_PATH_STATUS_SPECIFIC_TWO else
+                constants.HostStatus.OFFLINE,
+                'ip_address': host_iscsi.get('address')
+            }
+            host_list.append(host_d)
+
+    def get_data(self, command):
+        host_list = []
+        host_str = self.cli_handler.exec_command(command)
+        block = True
+        length_list = []
+        if host_str:
+            host_arr = host_str.strip().replace('\r', '').split('\n')
+            for host_row_str in host_arr:
+                if not host_row_str or \
+                        consts.CLI_STR in host_row_str:
+                    continue
+                if consts.SPECIAL_CHARACTERS_TWO in host_row_str:
+                    identify_list = host_row_str.split()
```

Comment:
These three sentences can be combined using a list derivation expression, which is more efficient. 
    length_list.extend([len(identify) for identify in host_row_str.split()])
![image](https://user-images.githubusercontent.com/59107077/153858482-42a27739-b853-4f61-a5a8-0397f1c4606e.png)

Reply:
Has been modified